services/QuillLMS/client/app/bundles/Evidence/components/studentView/container.tsx

```diff
@@ -103,6 +103,7 @@ export const StudentViewContainer = ({ dispatch, session, isTurk, location, acti
   }, [hasStartedReadPassageStep]);
 
   React.useEffect(() => {
+    dispatch(getTopicOptimalInfo(activityUID))
```

Comment:
Curious why you opted for the redux-y `dispatch` pattern here, when `fetch + useState` or `useReducer` would have accomplished the same thing. 

I'm not a redux expert, but it seems like hooks would require less code in this case.

Reply:
This is a totally reasonable point.  It may be a mistake, but my thought here was to follow the existing pattern already established in this file.  (Specifically, I'm copying the logic used in the `getActivity` action in this file already, since that's the other API call that gets made.)  I'm not sure this is the pattern we'd use if we were writing this component from scratch now, and would be willing to entertain the idea that we should build new components in the current state-of-the-art rather than re-using old approaches for new components.  Thoughts?

Reply:
IMO keeping the existing pattern here is fine, since you already have a PR hot and ready. 

In the future, I encourage us to prefer hooks over redux, in cases where we don't need complex reducer logic (which, for this app, is most cases).

Reply:
Yes, agreed with @happythenewsad that we should be opting for hooks where possible in general - I think this app was written before we were really using them and we just haven't updated old code to move towards that pattern. I also agree that it isn't something that needs to change for this PR, but makes sense to keep in mind in the future.